### ISSUE TITLE
Application error: a client-side exception has occurred (developer guidance).

### ISSUE THREAD

**Issue by colawithsauce** [org-roam/org-roam-ui#61]
# Issue
In `*httpd*` buffer, I get:
```
(stop "Fri Aug 13 08:27:43 2021")
(start "Fri Aug 13 08:27:45 2021")
(request
 (date "Fri, 13 Aug 2021 03:57:46 GMT")
 (address "127.0.0.1")
 (get "/sw.js")
 (headers
  ("GET" "/sw.js" "HTTP/1.1")
  ("Host" "localhost:35901")
  ("Connection" "keep-alive")
  ("Cache-Control" "max-age=0")
  ("User-Agent" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.25 Safari/537.36")
  ("Accept" "*/*")
  ("Service-Worker" "script")
  ("Sec-Fetch-Site" "same-origin")
  ("Sec-Fetch-Mode" "same-origin")
  ("Sec-Fetch-Dest" "serviceworker")
  ("Referer" "http://localhost:35901/")
  ("Accept-Encoding" "gzip, deflate, br")
  ("Accept-Language" "en-US,en;q=0.9")
  ("If-None-Match" "\"7f5c2115c2377148\"")
  ("If-Modified-Since" "Thu, 12 Aug 2021 17:34:28 GMT")
  ("Content" "")))
(file "/home/colawithsauce/.emacs.d/.local/straight/build-28.0.50/org-roam-ui/out/sw.js" not-modified)
(request
 (date "Fri, 13 Aug 2021 03:57:46 GMT")
 (address "127.0.0.1")
 (get "/workbox-ea903bce.js")
 (headers
  ("GET" "/workbox-ea903bce.js" "HTTP/1.1")
  ("Host" "localhost:35901")
  ("Connection" "keep-alive")
  ("Cache-Control" "max-age=0")
  ("User-Agent" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.25 Safari/537.36")
  ("Accept" "*/*")
  ("Sec-Fetch-Site" "same-origin")
  ("Sec-Fetch-Mode" "no-cors")
  ("Sec-Fetch-Dest" "script")
  ("Referer" "http://localhost:35901/sw.js")
  ("Accept-Encoding" "gzip, deflate, br")
  ("Accept-Language" "en-US,en;q=0.9")
  ("If-None-Match" "\"79cad6c0758a52dd\"")
  ("If-Modified-Since" "Thu, 12 Aug 2021 17:34:28 GMT")
  ("Content" "")))
(file "/home/colawithsauce/.emacs.d/.local/straight/build-28.0.50/org-roam-ui/out/workbox-ea903bce.js" not-modified)
(request
 (date "Fri, 13 Aug 2021 03:57:46 GMT")
 (address "127.0.0.1")
 (get "/_next/static/chunks/4.2dee5d830195ddd06029.js")
 (headers
  ("GET" "/_next/static/chunks/4.2dee5d830195ddd06029.js" "HTTP/1.1")
  ("Host" "localhost:35901")
  ("Connection" "keep-alive")
  ("Pragma" "no-cache")
  ("Cache-Control" "no-cache")
  ("User-Agent" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.25 Safari/537.36")
  ("Accept" "*/*")
  ("Sec-Fetch-Site" "same-origin")
  ("Sec-Fetch-Mode" "cors")
  ("Sec-Fetch-Dest" "empty")
  ("Referer" "http://localhost:35901/sw.js")
  ("Accept-Encoding" "gzip, deflate, br")
  ("Accept-Language" "en-US,en;q=0.9")
  ("Content" "")))
(error 404 nil)
(stop "Fri Aug 13 08:30:24 2021")
(start "Fri Aug 13 08:30:26 2021")
(connection "127.0.0.1")
(request
 (date "Fri, 13 Aug 2021 04:00:29 GMT")
 (address "127.0.0.1")
 (get "/sw.js")
 (headers
  ("GET" "/sw.js" "HTTP/1.1")
  ("Host" "localhost:35901")
  ("Connection" "keep-alive")
  ("Cache-Control" "max-age=0")
  ("User-Agent" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.25 Safari/537.36")
  ("Accept" "*/*")
  ("Service-Worker" "script")
  ("Sec-Fetch-Site" "same-origin")
  ("Sec-Fetch-Mode" "same-origin")
  ("Sec-Fetch-Dest" "serviceworker")
  ("Referer" "http://localhost:35901/")
  ("Accept-Encoding" "gzip, deflate, br")
  ("Accept-Language" "en-US,en;q=0.9")
  ("If-None-Match" "\"7f5c2115c2377148\"")
  ("If-Modified-Since" "Thu, 12 Aug 2021 17:34:28 GMT")
  ("Content" "")))
(file "/home/colawithsauce/.emacs.d/.local/straight/build-28.0.50/org-roam-ui/out/sw.js" not-modified)
(request
 (date "Fri, 13 Aug 2021 04:00:29 GMT")
 (address "127.0.0.1")
 (get "/workbox-ea903bce.js")
 (headers
  ("GET" "/workbox-ea903bce.js" "HTTP/1.1")
  ("Host" "localhost:35901")
  ("Connection" "keep-alive")
  ("Cache-Control" "max-age=0")
  ("User-Agent" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.25 Safari/537.36")
  ("Accept" "*/*")
  ("Sec-Fetch-Site" "same-origin")
  ("Sec-Fetch-Mode" "no-cors")
  ("Sec-Fetch-Dest" "script")
  ("Referer" "http://localhost:35901/sw.js")
  ("Accept-Encoding" "gzip, deflate, br")
  ("Accept-Language" "en-US,en;q=0.9")
  ("If-None-Match" "\"79cad6c0758a52dd\"")
  ("If-Modified-Since" "Thu, 12 Aug 2021 17:34:28 GMT")
  ("Content" "")))
(file "/home/colawithsauce/.emacs.d/.local/straight/build-28.0.50/org-roam-ui/out/workbox-ea903bce.js" not-modified)
(request
 (date "Fri, 13 Aug 2021 04:00:29 GMT")
 (address "127.0.0.1")
 (get "/_next/static/chunks/4.2dee5d830195ddd06029.js")
 (headers
  ("GET" "/_next/static/chunks/4.2dee5d830195ddd06029.js" "HTTP/1.1")
  ("Host" "localhost:35901")
  ("Connection" "keep-alive")
  ("Pragma" "no-cache")
  ("Cache-Control" "no-cache")
  ("User-Agent" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.25 Safari/537.36")
  ("Accept" "*/*")
  ("Sec-Fetch-Site" "same-origin")
  ("Sec-Fetch-Mode" "cors")
  ("Sec-Fetch-Dest" "empty")
  ("Referer" "http://localhost:35901/sw.js")
  ("Accept-Encoding" "gzip, deflate, br")
  ("Accept-Language" "en-US,en;q=0.9")
  ("Content" "")))
(error 404 nil)
```
# environment
1. emacs-version: 28.0.50, doom-emacs develop branch.
2. Distro: Archlinux
3. DE: Gnome40

**Comment by baconnette** [comment]
I also encountered the same error; it only happens when I toggle to 3d though. I just turned it off and it went back to normal. I'm in Macos Catalina, Emacs v27.2, Firefox v91

**Comment by MingleiYang** [comment]
I have the same issue of (error 404 nil). and there is nothing shown in the browser but black page.

**Comment by ThomasFKJorna** [comment]
This is a common problem and totally our bad! Please go to either Application > Local Storage or Storage > Local Storage in your Devtools (opened by pressing Ctrl+Shift+I or F12) and delete all the keys you find there. This will reset your settings but should solve your problem.

**Comment by sergiotaerri** [comment]
> This is a common problem and totally our bad! Please go to either Application > Local Storage or Storage > Local Storage in your Devtools (opened by pressing Ctrl+Shift+I or F12) and delete all the keys you find there. This will reset your settings but should solve your problem.

To me the error persists even after cleaning storage. Tried recompiling the package a few days ago and no success.

### environment
    emacs-version: 27.2, doom-emacs develop branch.
    Distro: Manjaro

**Comment by ThomasFKJorna** [comment]
> To me the error persists even after cleaning storage. Tried recompiling the package a few days ago and no success.

Has `org-roam-ui` worked for you in the past?

**Comment by serycjon** [comment]
It would be nice to show node labels "more intelligently". Now we can have all labels shown or no labels shown, based on the "label appearance scale". It would be really cool, if it worked something like maps, where you see only the most important labels when zoomed out (big cities) and as you zoom in, more labels (villages, street names, ...) start appearing.
This could be done based on number of in/outcoming links, or perhaps user-specifiable node-importance function.

**Comment by sergiotaerri** [comment]
> > To me the error persists even after cleaning storage. Tried recompiling the package a few days ago and no success.
> 
> Has `org-roam-ui` worked for you in the past?

Yes, It started to present this error after I did a `doom upgrade` this weekend.

**Comment by ThomasFKJorna** [comment]
That is very unfortunate, sorry! You mentioned you tried recompiling, did that include pulling the latest version? Else I'm not really sure what could be wrong short of 
- You not running org-roam-ui-mode
- Not having deleted *all* the keys
- Somehow pulling a bad commit

Could you show me the console log of your browser?

**Comment by sergiotaerri** [comment]
Of course,  the package is great so there's no need to be sorry! The recipe on my `packages.el` looks right, but I haven't checked the actual package folder in my machine to double check the version yet. Regardless, here's the log:
```
localhost/:1 GET http://localhost:35901/_next/static/r6rkJJOuqyQnmFLmB7qNo/_buildManifest.js net::ERR_ABORTED 404 (Not Found)
localhost/:1 GET http://localhost:35901/_next/static/r6rkJJOuqyQnmFLmB7qNo/_ssgManifest.js net::ERR_ABORTED 404 (Not Found)
localhost/:1 GET http://localhost:35901/_next/static/chunks/661-90187de4aa66e1891621.js net::ERR_ABORTED 404 (Not Found)
localhost/:1 GET http://localhost:35901/_next/static/chunks/pages/_app-803053ad060c5bee8d21.js net::ERR_ABORTED 404 (Not Found)
localhost/:1 GET http://localhost:35901/_next/static/chunks/pages/index-71a61839b1442c74b04b.js net::ERR_ABORTED 404 (Not Found)
localhost/:1 GET http://localhost:35901/_next/static/chunks/main-07c09d7ebdafc5ed9a99.js net::ERR_ABORTED 404 (Not Found)
```


Here's also the network screen:

![image](https://user-images.githubusercontent.com/48693028/129939435-30b2546f-dcd9-47f1-a8d0-81233e33cbf0.png)

**Comment by jueqingsizhe66** [comment]
Same error!

**Comment by ThomasFKJorna** [comment]
These issues all seem to have to do with the service worker. I have removed it (I added it for possible PWA support but it was not as straightforward as I'd hoped) so this might be fixed. However, pulling a new version probably won't remove the old service worker. Please open the Developer Options (Ctrl + Shift + I) on the page and go to
"Storage > Cache Storage" on firefox and delete everything there, or
"Application > Cache > Cache Storage" on Chromium and delete everything there.

Please upgrade to the latest version before you do this, otherwise ORUI will just add another service worker (although that will probably still resolve the issue, but only temporarily)

**Comment by jueqingsizhe66** [comment]
it works in firefox,  thanks for sharing.

**Comment by pinacle2000** [comment]
> These issues all seem to have to do with the service worker. I have removed it (I added it for possible PWA support but it was not as straightforward as I'd hoped) so this might be fixed. However, pulling a new version probably won't remove the old service worker. Please open the Developer Options (Ctrl + Shift + I) on the page and go to
> "Storage > Cache Storage" on firefox and delete everything there, or
> "Application > Cache > Cache Storage" on Chromium and delete everything there.
> 
> Please upgrade to the latest version before you do this, otherwise ORUI will just add another service worker (although that will probably still resolve the issue, but only temporarily)

Got same issue in Windows. My Edge "Application > Cache > Cache Storage" contained only one item and nothing changed after I deleted it. 

My ORUI worked fine before I did an upgrade yesterday (haven't done so a few weeks).

**Comment by pinacle2000** [comment]
> > These issues all seem to have to do with the service worker. I have removed it (I added it for possible PWA support but it was not as straightforward as I'd hoped) so this might be fixed. However, pulling a new version probably won't remove the old service worker. Please open the Developer Options (Ctrl + Shift + I) on the page and go to
> > "Storage > Cache Storage" on firefox and delete everything there, or
> > "Application > Cache > Cache Storage" on Chromium and delete everything there.
> > Please upgrade to the latest version before you do this, otherwise ORUI will just add another service worker (although that will probably still resolve the issue, but only temporarily)
> 
> Got same issue in Windows. My Edge "Application > Cache > Cache Storage" contained only one item and nothing changed after I deleted it.
> 
> My ORUI worked fine before I did an upgrade yesterday (haven't done so a few weeks).

The issue is gone after I upgrade the package today.

**Comment by ThomasFKJorna** [comment]
Great to hear that! I tried to fix this issue but forgot to tag it, glad to see it worked!

**Comment by RobinKD** [comment]
![Screenshot_20210927_170158](https://user-images.githubusercontent.com/28922613/134934489-b74cf136-e5ef-4c6f-888c-289650e498bb.png)
Hi,
I also get the same errors.
I have been trying to make it work with doom emacs (haven't been able to make it work yet, I started trying a few days ago). I'm loading org-roam v2 with doom, and unpin it before ORUI package declaration.
Error seems to be the same for me, whether it be firefox or chromium.
The http buffer also has (error 404 nil) as reply for the request for those files.

I tried with different ORUI commits, but no luck so far.
I have been following the instructions, so I don't know if there's something wrong with my config or not.

**Comment by thriveth** [comment]
OK, I found it myself, so for posterity: I opened Developer Tools with C-S-i, then navigated to Application > Local Storage or Storage > Local Storage > http://localhost:35901. In the bottom of the table there, there is a row which said 3d | true which I manually changed to "false", then reloaded the page, and it worked again. 

![image](https://user-images.githubusercontent.com/1553566/136696478-cacdc93b-2d0a-4cb4-981f-2f506e029ca8.png)
